### PR TITLE
Dependencies update 

### PR DESCRIPTION
**Pull request by Smartappli**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by Smartappli** [comment]
@kiendang can you review this PR plz?

**Comment by Smartappli** [comment]
https://github.com/OpenMined/PySyft/pull/8896

**Comment by Smartappli** [comment]
@yashgorana can you review plz?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -181,6 +181,11 @@ repos:
       - id: prettier
         exclude: ^(packages/grid/helm|packages/grid/frontend/pnpm-lock.yaml|packages/syft/tests/mongomock)
 
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.16.0
+    hooks:
+      - id: pyupgrade
+
```
Comment (kiendang) [inline .pre-commit-config.yaml]: We already enabled pyupgrade rules in ruff so this is redundant. We specifically don't run automated linting on `tests/mongomock` because it's a fork.

File: tox.ini [hunk 2]
```diff
@@ -232,7 +232,7 @@ changedir = {toxinidir}/packages/syft
 deps =
     {[testenv:syft]deps}
 commands =
-    bandit -r src
+    bandit -r packages
```
Comment (kiendang) [inline tox.ini]: why this change?

File: scripts/patch_hosts.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def path(self) -> Path:
             old_hosts = "/private/etc/hosts"
             p = new_hosts if os.path.exists(new_hosts) else old_hosts
         elif Platform.windows():
-            p = "C:\Windows\System32\drivers\etc\hosts"
+            p = r"C:\Windows\System32\drivers\etc\hosts"
```
Comment (kiendang) [inline scripts/patch_hosts.py]: I don't think using `r""` here is correct.

**Review by kiendang — CHANGES_REQUESTED** [review]

File: scripts/patch_hosts.py (hunk 3 above)
Comment (Smartappli) [reply scripts/patch_hosts.py]: reverted

File: tox.ini (hunk 2 above)
Comment (Smartappli) [reply tox.ini]: reverted

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Smartappli) [reply .pre-commit-config.yaml]: reverted

File: scripts/patch_hosts.py (hunk 3 above)
Comment (Smartappli) [reply scripts/patch_hosts.py]: reverted

**Comment by madhavajay** [comment]
Hi @Smartappli can we chat on Slack?

**Comment by Smartappli** [comment]
Yes of course

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________
